server/leafnode.go

```diff
@@ -2125,11 +2125,11 @@ func (acc *Account) updateLeafNodes(sub *subscription, delta int32) {
 		}
 		// Check to make sure this sub does not have an origin cluster that matches the leafnode.
 		ln.mu.Lock()
-		skip := (cluster != _EMPTY_ && cluster == ln.remoteCluster()) || (delta > 0 && !ln.canSubscribe(subject))
 		// If skipped, make sure that we still let go the "$LDS." subscription that allows
-		// the detection of a loop.
-		if isLDS || !skip {
-			ln.updateSmap(sub, delta)
+		// the detection of loops as long as different cluster.
+		clusterDifferent := cluster != ln.remoteCluster()
+		if (isLDS && clusterDifferent) || (cluster == _EMPTY_ || clusterDifferent) && (delta <= 0 || ln.canSubscribe(subject)) {
```

Comment:
Did not review the whole PR, but the last `&& (delta <= 0 ...` is a bit confusing, meaning, the `if` statement looks like: `if A || B && C` and it would be nice to add parenthesis to make sure of the intent (is it `if A || (B && C)` or `if (A || B) && C`?

Reply:
Its A || B && C. The negative of the original I changed just to inline and not have to evaluate everything.

Reply:
I will make more clear for the reader.